x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -372,16 +401,21 @@ func (k Keeper) FulfillsMinStake(ctx sdk.Context, consumerId string, providerAdd
 }
 
 // ComputeNextValidators computes the validators for the upcoming epoch based on the currently `bondedValidators`.
-func (k Keeper) ComputeNextValidators(ctx sdk.Context, consumerId string, bondedValidators []stakingtypes.Validator, minPowerToOptIn int64) []types.ConsensusValidator {
+func (k Keeper) ComputeNextValidators(
+	ctx sdk.Context,
+	consumerId string,
+	bondedValidators []stakingtypes.Validator,
+	powerShapingParameters types.PowerShapingParameters,
```

Comment:
If I recall correctly, in some previous PR, it was mentioned that we shouldn't pass `powerShapingParameters` or `initializationParameters` if the `consumerId` is also passed as an argument because we can retrieve those from `consumerId`. I prefer it this way.
Do we have a rule on when to pass and when not?

Reply:
The rule is to try to avoid accessing the state multiple times for no reason. In this case, the powerShapingParameters are available everytime when calling ComputeNextValidators, so it makes sense to pass it. The consumerId is needed for the other methods, i.e., FilterValidators and CanValidateChain.